### ISSUE TITLE
All installed plugins should be loaded by qgis_process if running headless

### ISSUE THREAD

**Issue by elliots** [qgis/QGIS#41218]
**Feature description.**

I'm attempting to install the [UMEP for Processing](https://plugins.qgis.org/plugins/processing_umep/) plugin into a docker container to be used via qgis_process. 

After unzipping the plugin into the plugins directory, it appears when running `qgis_process plugins` but is not "loaded".

```
[root@c4da1ff0ffde /]# qgis_process plugins
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
<string>:1: DeprecationWarning: setapi() is deprecated
/usr/lib/python3.9/site-packages/qgis/utils.py:792: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  mod = _builtin_import(name, globals, locals, fromlist, level)
Problem with GRASS installation: GRASS was not found or is not correctly installed
Problem with SAGA installation: SAGA was not found or is not correctly installed
Available plugins
(* indicates loaded plugins which implement Processing providers)

  processing_umep
* processing
```

I see that there was a change forcing the `processing` plugin to always be loaded, but only that plugin.

If qgis_process is running headless, should all plugins be loaded automatically?

Alternatively, if other configuration is required to make it load, the help message could be clearer (it isn't obvious if the plugin hasn't been loaded, or does not implement Processing provider).

**Additional context**

This is the dockerfile, in case its of any use:

```Dockerfile

FROM lopsided/archlinux

RUN pacman -Syy && pacman --noconfirm -S qgis python-pip python-pandas python-scipy unzip

ENV QT_QPA_PLATFORM offscreen

RUN mkdir -p ~/.local/share/QGIS/QGIS3/profiles/default/QGIS && \
  echo "[PythonPlugins]\nprocessing=true\nprocessing_umap=true" > ~/.local/share/QGIS/QGIS3/profiles/default/QGIS/QGIS3.ini && \
  curl https://plugins.qgis.org/plugins/processing_umep/version/0.5/download/ --output umep-processing.zip && \
  mkdir -p ~/.local/share/QGIS/QGIS3/profiles/default/python/plugins && \
  unzip umep-processing.zip -d ~/.local/share/QGIS/QGIS3/profiles/default/python/plugins && \
  rm umep-processing.zip

```

**Comment by nyalldawson** [comment]
The original intention here was to add a command to qgis_process to permit enabling / disabling of plugins via the command line, but it hasn't yet be added.

Eg.

    qgis_process plugin enable processing_umep
    qgis_process plugin disable processing_umep

**Comment by elliots** [comment]
I'm adding it to ~/.local/share/QGIS/QGIS3/profiles/default/QGIS/QGIS3.ini which i believe should get past the  `mPythonUtils->isPluginEnabled` call at https://github.com/qgis/QGIS/blob/04b9c6ab00bb130e590863244b8de64c0f111ae1/src/process/qgsprocess.cpp#L440 but doesn't.

**Comment by elliots** [comment]
If anyone is looking for a dodgy way to make it work for now... this dockerfile will give you a `qgis_process` that loads all plugins

```Dockerfile
FROM qgis/qgis:release-3_16

# build patched qgis_process that loads all plugins
RUN sed -i 's/plugin == QLatin1String( "processing" ) || ( mPythonUtils->isPluginEnabled( plugin ) && mPythonUtils->pluginHasProcessingProvider( plugin ) )/mPythonUtils->pluginHasProcessingProvider( plugin )/' /QGIS/src/process/qgsprocess.cpp
RUN cd /QGIS/build && ninja qgis_process && cp output/bin/qgis_process /usr/bin
```

**Comment by beedawg1985** [comment]
> If anyone is looking for a dodgy way to make it work for now... this dockerfile will give you a `qgis_process` that loads all plugins
> 
> ```dockerfile
> FROM qgis/qgis:release-3_16
> 
> # build patched qgis_process that loads all plugins
> RUN sed -i 's/plugin == QLatin1String( "processing" ) || ( mPythonUtils->isPluginEnabled( plugin ) && mPythonUtils->pluginHasProcessingProvider( plugin ) )/mPythonUtils->pluginHasProcessingProvider( plugin )/' /QGIS/src/process/qgsprocess.cpp
> RUN cd /QGIS/build && ninja qgis_process && cp output/bin/qgis_process /usr/bin
> ```

I'm struggling with this workaround at the moment because I'm trying to set up the dockerised qgis_process on an SSH server. The process of forwarding X11 locally from a docker container on a remote server is complicated (see https://stackoverflow.com/questions/48235040/run-x-application-in-a-docker-container-reliably-on-a-server-connected-via-ssh-w) and so I can't launch the QGIS GUI to install the plug-ins (and qgis_process also seems to need X11 forwarding / GUI to be working too?). 

Is the updated qgis_process available in any QGIS releases yet?

**Comment by agiudiceandrea** [comment]
> Is the updated qgis_process available in any QGIS releases yet?

@beedawg1985 if you are referring to https://github.com/qgis/QGIS/pull/41259, then it's available in QGIS >= 3.16.5 and >= 3.18.0.

**Comment by beedawg1985** [comment]
Great; I was running an older version of QGIS accidentally. Thank you I will check it out
